modules/margin-liquidity-pools/src/lib.rs

```diff
@@ -167,12 +170,27 @@ decl_module! {
 			Self::deposit_event(RawEvent::SetEnabledTrades(who, pool_id, pair, enabled));
 		}
 
-		pub fn update_swap(origin, pool_id: T::LiquidityPoolId, pair: TradingPair, rate: Fixed128) {
+		pub fn set_swap_rate(origin, pair: TradingPair, long_rate: Fixed128, short_rate: Fixed128) {
+			T::UpdateOrigin::try_origin(origin)
+				.map(|_| ())
+				.or_else(ensure_root)?;
+			ensure!(long_rate >= T::MaxSwap::get().0 && long_rate <= Fixed128::zero(), Error::<T>::SwapRateTooLow);
+			ensure!(short_rate >= Fixed128::zero() && short_rate <= T::MaxSwap::get().1, Error::<T>::SwapRateTooHigh);
+			SwapRates::insert(pair, (long_rate, short_rate));
+			Self::deposit_event(RawEvent::SwapRateUpdated(pair, long_rate, short_rate));
+		}
+
+		pub fn set_additional_swap(origin, pool_id: T::LiquidityPoolId, pair: TradingPair, rate: Fixed128) {
```

Comment:
The additional swap is per pool, so the trading pair parameter is not needed. And the checks should be made on all trading pairs supported by this pool.